tools/fmt_diff/fmt_diff.ml

```diff
@@ -0,0 +1,118 @@
+open Rresult
+
+(** [run_ocamlformat cmd ~ctx ~file_in] runs the ocamlformat command [cmd]
+    on the input file [file_in] with the configuration context of file [ctx]
+    and returns the path to the output in file. *)
+let run_ocamlformat cmd ~ctx ~file_in =
+  Bos.OS.File.tmp "%s" >>= fun path_out ->
+  Bos.Cmd.(cmd % Fpath.to_string file_in
+           % "--name" % Fpath.to_string ctx % "--disable-outside-project"
+           % "-o" % Fpath.to_string path_out)
+  |> Bos.OS.Cmd.run_status
+  |> function
+  | Ok (`Exited 0) -> Ok path_out
+  | Ok _ ->
+      R.error_msg ("ocamlformat on " ^ Fpath.to_string file_in ^ " failed!")
+  | Error e -> Error e
+
+(** [diff_in_ctx ~ctx ~old_file ~new_file] runs the diff of files [new_file]
+    and [new_file] in the context of file [ctx], so that they both inherit
+    the same configuration. *)
+let diff_in_ctx ~ctx ~old_file ~new_file =
+  Bos.OS.Cmd.resolve (Bos.Cmd.v "ocamlformat") >>= fun fmt ->
+  run_ocamlformat fmt ~ctx ~file_in:old_file >>= fun tmp_1 ->
+  run_ocamlformat fmt ~ctx ~file_in:new_file >>= fun tmp_2 ->
+  Bos.OS.Cmd.must_exist (Bos.Cmd.v "diff") >>= fun diff ->
```

Comment:
As someone who has the following in my .gitconfig:
```
[diff]
        tool = patdiff
[difftool "patdiff"]
	cmd = patdiff-git-wrapper $BASE $LOCAL 0 0 $REMOTE 0 0
```
I would be much happier if there was some way to similarly configure `fmt-diff` (other than PATH shadowing hacks).

Reply:
I would like to avoid ocamlformat-diff specific configuration files if possible, how about an environment variable $ocamlformat_diff_default=patdiff ?

Reply:
By the way the integration with patdiff works fine with my last commit